### PR TITLE
final fix on github mixup

### PR DESCRIPTION
**Pull request by RebeccaStankus**
This PR incorporates a new checkout stand model. Recycle zone and Scanner zone will only search for overlays while an avatar is in the checkout zone and there is an updated checkout zone script that calls methods from recycle and scanner when an avatar enters the zone, and an updated JSON for the entire checkout area including mirror, checkout stand, and all relevant zones. Recycle zone now deletes the item on the avatar when it's matching overlay is deleted. Tablet will redirect to wallet app if the user's wallet is not set up and authenticated. An item's position on the avatar will be stored upon scanning the item so that it can be rezzed in the same spot. This allows for adjustment inside the checkout zone. There will be more fine tuning of the placement of overlays when we have the final model, but for now, we can test that items are coming in relatively where we would like them to be. The scanner will not delete overlays anymore, but will move them to the platform on the right instead. Changes were also made to the position of the tablet to accommodate the new model. A new miniature checkout was added for small avatars. Highlighting is added, but turned off for now.

TEST PLAN-------------------------------------------
1-Have two users go to dev-content, A&J store. User one is normal size, and User two is small...about the size of Robimo. (Note that Robimo does not have a "Head" joint and can not try on hats or other things attached to the head...)
2-User one will need to open and close the tablet before proceeding to get accurate results.
3-User one enters checkout having set up wallet in previous session, but not having authenticated during this session
4-Upon entering checkout area, tablet will open to authentication page.
5-Repeat with each user having set up wallet and authenticated during the current session.
6-Upon entering checkout area, tablet will open to checkout instructions page.
7-Each user tries on a the monster beanie, cowboy hat, and purple scarf.
8-User one goes to the checkout zone and stands in front of the middle checkout stand.
9-User one should see repeating messages in Debug Window about the scanner and recycle bin searching for items.
10-User two should not see these messages.
11-User one enters normal sized checkout while User two enters the miniature checkout.
12-Each user grabs an overlay, places it in the recycle bin, and releases it.
13-The overlay and its matching item on the avatar should both disappear.
14-The overlay should move to the platform on the right.
15-The tablet should redirect to the marketplace page of the item in scanner.
16-Each user purchases the item and then clicks "Wear it"
17-Item should briefly disappear and then reappear in the same position it was in when its matching overlay was scanned.
18-The item should very briefly disappear and reappear in the same position.
19-Upon User one leaving the checkout area, the messages about the scanner and recycle bin should stop.

### CODE REVIEW

File: DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js [hunk 1]
```diff
@@ -8,77 +8,290 @@
...
+                        if (Overlays.getProperty(overlayInScanner, 'parentID') === tableID) { 
+                            // if item in Scanner is not being grabbed anymore
+                            name = Overlays.getProperty(overlayInScanner, 'name');
+                            if (highlightToggle) {
+                                Selection.removeFromSelectedItemsList(LIST_NAME, "entity", matchingEntity);
+                                Selection.removeFromSelectedItemsList(LIST_NAME, "overlay", overlayInScanner);
+                                prevID = 0;
+                            }
+                            var marketplaceID = name.substr(OVERLAY_PREFIX.length, OVERLAY_PREFIX.length + 35);
+                            print("mp id " + marketplaceID);
```
Comment (misslivirose) [inline DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js]: remove print statement, 35 magic number

File: DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js [hunk 2]
```diff
@@ -8,77 +8,290 @@
...
+                            var goToURL = MARKET_PLACE_ITEM_URL_PREFIX + "/items/" + marketplaceID;
+                            var entityProperties = Entities.getEntityProperties(matchingEntity, 
+                                ['localPosition', 'localRotation', 'dimensions', 
+                                    'parentJointIndex', 'marketplaceID']);
```
Comment (misslivirose) [inline DomainContent/AvatarStore/CheckoutZone/checkoutScanner.js]: this line doesn't have to be intended

File: DomainContent/AvatarStore/CheckoutZone/checkoutZone.js [hunk 3]
```diff
@@ -10,40 +10,49 @@
...
     var replicaList = [];
+    var button;
+    var recycleBinID;
+    var scannerZone;
     var replicaStoredTransforms = {};
     var left = true;
-    var button;
-
+    var middle = false;
+
```
Comment (misslivirose) [inline DomainContent/AvatarStore/CheckoutZone/checkoutZone.js]: Some extra spaces in this file?

File: DomainContent/AvatarStore/CheckoutZone/checkoutZone.js [hunk 4]
```diff
@@ -281,11 +314,12 @@
...
+  
     _this.unload = function() {
-        // sure you leave the entity if you're still in there
+        // make sure you leave the entity if you're still in there
         if (isInZone) {
             _this.leaveEntity();
         }
     };
 });
+
```
Comment (misslivirose) [inline DomainContent/AvatarStore/CheckoutZone/checkoutZone.js]: newline?

File: DomainContent/AvatarStore/CheckoutZone/MiniCheckout/checkoutRecycle.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            if (highlightToggle) {
+                highlightConfig["selectionName"] = LIST_NAME; 
+                Selection.clearSelectedItemsList(LIST_NAME);
+                HIGHLIGHT.changeHighlight2(highlightConfig);
+            }
+            tableID = Entities.getEntityProperties(entityID, 'parentID').parentID;
+        },
+        enterCheckout: function() {
+            interval = Script.setInterval(function() {
+                print("Recycle is searching too...");
```
[the remark this answers is not in the document]
Comment (RebeccaStankus) [reply DomainContent/AvatarStore/CheckoutZone/MiniCheckout/checkoutRecycle.js]: it was for testing

File: DomainContent/AvatarStore/CheckoutZone/MiniCheckout/checkoutRecycle.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+                            overlayInBin = null;
+                        }
+                    }
+                } else if (overlays.length > 0 && !overlayInBin) { // check new overlays in bin
+                    overlays.forEach(function(overlayID) {
+                        var name = Overlays.getProperty(overlayID, 'name');
+                        if (name.indexOf(OVERLAY_PREFIX) !== -1) {
+                            var nearbyEntities = Entities.findEntities(MyAvatar.position, AVATAR_SEARCH_RADIUS);
+                            nearbyEntities.forEach(function(entityID) {
+                                var userDataString =
```
[the remark this answers is not in the document]
Comment (RebeccaStankus) [reply DomainContent/AvatarStore/CheckoutZone/MiniCheckout/checkoutRecycle.js]: Yes...it made the line129 chars...

**Pull request merged**